File: Python/그리디 알고리즘/1946.py
import sys
input = sys.stdin.readline
T = int(input())

for _ in range(T):
    N = int(input())
    arr = list()
    res = 1
    
    for i in range(N):
        arr.append(tuple(map(int,input().split())))
    arr.sort() #서류 성적 순으로 정렬 ==> 서류 성적 1등의 면접 성적보다 나머지 사람들의 면접 성적이 높지 않다면 제외
    
    m = arr[0][1]
    for i in range(1, N):
        if arr[i][1] < m:
            res += 1
            m = arr[i][1]

    print(res)

File: BOJ/Kotlin/그리디 알고리즘/1946.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val T = readLine().toInt()

    for (i in 0 until T){
        val N = readLine().toInt()
        val arr = arrayOfNulls<Grade>(N)
        var res = 1

        for (m in 0 until N){
            val (paper, com) = readLine().split(' ').map { it.toInt() }
            arr[m] = Grade(paper, com)
        }
        arr.sortWith(compareBy { it!!.paper })

        var min = arr[0]!!.communication
        for (n in 1 until N){
            if (arr[n]!!.communication < min){
                res += 1
                min = arr[n]!!.communication
            }
        }
        println(res)
    }
}

data class Grade(val paper: Int, val communication: Int)